File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03239/p03239_num8_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int N, T;
    cin >> N >> T;

    vector<pair<int, int>> routes(N);
    for (int i = 0; i < N; i++) {
        cin >> routes[i].first >> routes[i].second;
    }

    sort(routes.begin(), routes.end(), [](const pair<int, int>& a, const pair<int, int>& b) {
        return a.second < b.second;
    });

    int minCost = -1;
    for (int i = 0; i < N; i++) {
        if (routes[i].second <= T) {
            minCost = routes[i].first;
            break;
        }
    }

    if (minCost == -1) {
        cout << "TLE" << endl;
    } else {
        cout << minCost << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03239/p03239_num8_parsed.py
def main():
    N, T = map(int, input().split())
    routes = []
    for _ in range(N):
        c, t = map(int, input().split())
        routes.append((c, t))
    
    routes.sort(key=lambda x: x[0])
    
    min_cost = float('inf')
    for c, t in routes:
        if t <= T:
            min_cost = min(min_cost, c)
    
    if min_cost == float('inf'):
        print("TLE")
    else:
        print(min_cost)

if __name__ == "__main__":
    main()